include/OpenColorIO/OpenColorIO.h

```diff
@@ -649,9 +650,95 @@ class OCIOEXPORT Config
 
+    /** @} */
+
+    /**
+     * \defgroup Methods related to the Virtual Display.
+     * @{
+     *
+     *  ...  (See descriptions for the non-virtual methods above.)
+     *
+     * The virtual display is the way to incorporate the ICC monitor profile for a user's display
+     * into OCIO. The views that are defined for the virtual display are the views that are used to
+     * create a new display for an ICC profile. They serve as a kind of template that lets OCIO
+     * know how to build the new display.
+     *
+     * Typically the views will define a View Transform and set the colorSpaceName to 
+     * "<USE_DISPLAY_NAME>" so that it will use the display color space with the same name as the
+     * display, in this case corresponding to the ICC profile.
+     *
+     */
+
+    void addVirtualDisplayView(const char * view,
+                               const char * viewTransformName,
+                               const char * colorSpaceName,
+                               const char * looks,
+                               const char * ruleName,
+                               const char * description);
+
+    void addVirtualDisplaySharedView(const char * sharedView);
+
+    /// Get the number of views associated to the virtual display.
+    int getVirtualDisplayNumViews(ViewType type) const noexcept;
+    /// Get the view name at a specific index.
+    const char * getVirtualDisplayView(ViewType type, int index) const noexcept;
+
+    const char * getVirtualDisplayViewTransformName(const char * view) const noexcept;
+    const char * getVirtualDisplayViewColorSpaceName(const char * view) const noexcept;
+    const char * getVirtualDisplayViewLooks(const char * view) const noexcept;
+    const char * getVirtualDisplayViewRule(const char * view) const noexcept;
+    const char * getVirtualDisplayViewDescription(const char * view) const noexcept;
+
+    /// Remove the view from the virtual display.
+    void removeVirtualDisplayView(const char * view) noexcept;
+
+    /// Clear the virtual display.
+    void clearVirtualDisplay() noexcept;
+
+    /**
+     * \brief Instantiate a new display from a virtual display, using the monitor name.
+     * 
+     * This method uses the virtual display to create an actual display for the given monitorName.
+     * The new display will receive the views from the virtual display.
+     *
+     * After the ICC profile is read, a display name will be created by combining the description
+     * text from the profile with the monitorName obtained from the OS. Use the SystemMonitors class
+     * to obtain the list of monitorName strings for the displays connected to the computer.
+     *
+     * A new display color space will also be created using the display name. It will have a
+     * from_display_reference transform that is a FileTransform pointing to the ICC profile.
+     *
+     * Any instantiated display color spaces for a virtual display are intended to be temporary
```

Comment:
This is cool.

Reply:
Thanks. That was a lot of work to correctly 'model' (with the community help during WG sessions) and implement the feature. But the benefit is clearly here.